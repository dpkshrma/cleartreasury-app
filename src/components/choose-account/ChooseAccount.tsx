import React from "react";
import { ChevronRightIcon } from "@heroicons/react/outline";

const ChooserAccount = ({ accounts, onAccountSelect }) => {
  const onClickHandler = (event, account) => {
    event.preventDefault();
    onAccountSelect(account);
  };

  return (
    <div className="w-screen h-screen flex flex-col space-y-6">
      <h1 className="bg-gray-200 text-center p-8 mt-14 text-4xl">
        Which account would you like to use?
      </h1>

      <div className="w-2/5 mx-auto">
        <h2 className="text-2xl">Account details</h2>
        <p className="text-lg text-gray-600">
          Your email address is associated with multiple accounts. Please select
          one to continue{" "}
        </p>
      </div>

      <ul className="space-y-2 w-2/5 mx-auto">
        {accounts.map((account) => (
          <a
            href="#"
            key={account.cli_reference}
            onClick={(e) => onClickHandler(e, account)}
            className="text-xl text-gray-600 bg-gray-200 p-4 flex justify-between items-center"
          >
            {account.cli_name}
            <ChevronRightIcon className="w-5 h-5" />
          </a>
        ))}
      </ul>
    </div>
  );
};

export default ChooserAccount;
