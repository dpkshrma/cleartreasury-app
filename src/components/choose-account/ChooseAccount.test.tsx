import * as React from "react";
import { render } from "../../../test/testUtils";
import * as Stories from "./ChooseAccount.stories";
import { fireEvent } from "@testing-library/react";

describe("ChooseAccount", () => {
  it("matches snapshot choose account component", () => {
    const onAccountSelect = jest.fn();

    const { asFragment, getByText } = render(
      <Stories.Default
        {...Stories.Default.args}
        onAccountSelect={onAccountSelect}
      />
    );

    fireEvent.click(getByText(Stories.accounts[0].cli_name).closest("a"));
    expect(onAccountSelect).toBeCalled();
    expect(asFragment()).toMatchSnapshot();
  });
});
