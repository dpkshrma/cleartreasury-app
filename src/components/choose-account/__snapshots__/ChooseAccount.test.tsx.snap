// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChooseAccount matches snapshot choose account component 1`] = `
<DocumentFragment>
  <div
    class="w-screen h-screen flex flex-col space-y-6"
  >
    <h1
      class="bg-gray-200 text-center p-8 mt-14 text-4xl"
    >
      Which account would you like to use?
    </h1>
    <div
      class="w-3/5 md:w-2/5 mx-auto"
    >
      <h2
        class="text-2xl"
      >
        Account details
      </h2>
      <p
        class="text-lg text-gray-600"
      >
        Your email address is associated with multiple accounts. Please select one to continue 
      </p>
    </div>
    <ul
      class="space-y-2 w-3/5 md:w-2/5 mx-auto"
    >
      <a
        class="bg-gray-200 p-4 flex justify-between items-center hover:bg-gray-300"
        href="#"
      >
        <div
          class="flex items-center space-x-2"
        >
          <svg
            class="w-16 h-16 flex-shrink-0 text-teal-400 font-normmal"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.121 17.804A13.937 13.937 0 0112 16c2.5 0 4.847.655 6.879 1.804M15 10a3 3 0 11-6 0 3 3 0 016 0zm6 2a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <div>
            <h3
              class="text-xl text-gray-600"
            >
              cli_name_1
            </h3>
            <span
              class="text-gray-400"
            >
              Personal account
            </span>
          </div>
        </div>
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 5l7 7-7 7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
      <a
        class="bg-gray-200 p-4 flex justify-between items-center hover:bg-gray-300"
        href="#"
      >
        <div
          class="flex items-center space-x-2"
        >
          <svg
            class="w-16 h-16 flex-shrink-0 text-teal-400 font-normmal"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M5.121 17.804A13.937 13.937 0 0112 16c2.5 0 4.847.655 6.879 1.804M15 10a3 3 0 11-6 0 3 3 0 016 0zm6 2a9 9 0 11-18 0 9 9 0 0118 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
          <div>
            <h3
              class="text-xl text-gray-600"
            >
              cli_name_2
            </h3>
            <span
              class="text-gray-400"
            >
              Personal account
            </span>
          </div>
        </div>
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M9 5l7 7-7 7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
    </ul>
  </div>
</DocumentFragment>
`;
